presto-docs/pom.xml

```diff
@@ -62,6 +62,36 @@
                             </arguments>
                         </configuration>
                     </execution>
+                    <execution>
+                        <id>generate-thrift-idl</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>java</goal>
+                        </goals>
+                        <configuration>
+                            <mainClass>com.facebook.swift.generator.swift2thrift.Main</mainClass>
+                            <arguments>
+                                <argument>com.facebook.presto.connector.thrift.api.PrestoThriftService</argument>
+                                <argument>-recursive</argument>
+                                <argument>-out</argument>
+                                <argument>${project.build.directory}/PrestoThriftService.thrift</argument>
+                            </arguments>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>validate-thrift-idl</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>exec</goal>
```

Comment:
At first I thought this was hacky, but it's actually quite clever. It will return an exit code based on if they are different and also output any differences. As long as it's available in the path, this will work. And diff behavior is specified in POSIX: http://pubs.opengroup.org/onlinepubs/009696799/utilities/diff.html